zebra-test/tests/transcript.rs

```diff
@@ -35,7 +41,42 @@ async fn transcript_errors_wrong_request() {
 
 #[tokio::test]
 async fn self_check() {
+    zebra_test::init();
+
     let t1 = Transcript::from(TRANSCRIPT_DATA.iter().cloned());
     let t2 = Transcript::from(TRANSCRIPT_DATA.iter().cloned());
     assert!(t1.check(t2).await.is_ok());
 }
+
+const TRANSCRIPT_DATA2: [(&str, Result<&str, ErrorChecker>); 4] = [
+    ("req1", Ok("rsp1")),
+    ("req2", Ok("rsp2")),
+    ("req3", Ok("rsp3")),
+    (
+        "req4",
+        Err(|e| {
+            if e.is_none() {
+                Err("this is bad")?;
+            }
+
+            let e = e.unwrap();
+
+            if e.to_string() == "this is bad" {
+                Ok(())
+            } else {
+                Err(e)
+            }
+        }),
+    ),
```

Comment:
This is a lot more unwieldy than the non-error case, so I think this is a signal that we're not defining the problem and modeling the solution correctly.  It's worth re-evaluating the problem space before continuing here.